### PR TITLE
Store Fixes

### LINKED ISSUE #181: Hirelings transactions on gamestore
## CAUTION: OUR SERVER IS UPDATED DAILY. MAKE SURE YOU ARE REPORTING SOMETHING ON THE LATEST VERSION OF THE SERVER.

**Describe the bug**
When buying a new Hireling on gamestore the player is being charged 3 times for the same transaction:
1. When choosing the hireling and clicking on "Buy" (and confirming on the next screen);
1.1. After that you're charged for the first time.
2. Now you're prompted to enter the Hireling's name
2.1. After providing it you're asked to buy it again, and that's when you get charged for the second and third times (I'm guessing you're charged one time when clicking ok after providing the hireling's name and a third one when clicking no "Buy" again).

**To Reproduce**
0. Make sure you have enough Tibia Coins for at least 3 hirelings (450 TC);
1. Go to gamestore;
2. Click on "House" and then on "Hirelings"
3. Choose "Hireling Male" or "Hireling Female"
4. Click on Buy and follow the flow
5. After ending go to History (there will be 3 transactions registered about the same hireling)
6. You can also see your TC balance.

**Expected behavior**
Just be charged one time for the hireling (probably just after providing its name).

**Screenshots**
https://snipboard.io/Doglhc.jpg

**Desktop (please complete the following information):**
 - OS: Windows
 - Version: 10

### LINKED ISSUE #1566: Hirelings transactions on gamestore
## CAUTION: OUR SERVER IS UPDATED DAILY. MAKE SURE YOU ARE REPORTING SOMETHING ON THE LATEST VERSION OF THE SERVER.

**Describe the bug**
When buying a new Hireling on gamestore the player is being charged 3 times for the same transaction:
1. When choosing the hireling and clicking on "Buy" (and confirming on the next screen);
1.1. After that you're charged for the first time.
2. Now you're prompted to enter the Hireling's name
2.1. After providing it you're asked to buy it again, and that's when you get charged for the second and third times (I'm guessing you're charged one time when clicking ok after providing the hireling's name and a third one when clicking no "Buy" again).

**To Reproduce**
0. Make sure you have enough Tibia Coins for at least 3 hirelings (450 TC);
1. Go to gamestore;
2. Click on "House" and then on "Hirelings"
3. Choose "Hireling Male" or "Hireling Female"
4. Click on Buy and follow the flow
5. After ending go to History (there will be 3 transactions registered about the same hireling)
6. You can also see your TC balance.

**Expected behavior**
Just be charged one time for the hireling (probably just after providing its name).

**Screenshots**
https://snipboard.io/Doglhc.jpg

**Desktop (please complete the following information):**
 - OS: Windows
 - Version: 10

### LINKED ISSUE #1738: XP boost only works once a day
xpboosts

### PR DESCRIPTION
**Pull request by FakeShinoda**
- Open categories when click in button (buy premium, prey slot button etc..)
- Hireling offers like global tibia
- Fix #1566 (charged 3 times)

Note: Hireling Outfits not included.

Credits to:
@andersonfaaria [sort hirelings offers]

### CODE REVIEW

File: data/migrations/5.lua [hunk 1]
```diff
@@ -1,3 +1,5 @@
 function onUpdateDatabase()
-    return false -- true = There are others migrations file | false = this is the last migration file
+    print("> Updating database to version 6 (sexId update)")
+	db.query([[UPDATE `players` SET `sex` = '0' WHERE `sex` = 2;]])
```
Comment (majestyotbr) [inline data/migrations/5.lua]:
```
db.query([[UPDATE `players` SET `sex` = '2' WHERE `sex` = 0;]])
```

File: data/modules/scripts/gamestore/gamestore.lua [hunk 2]
```diff
@@ -2275,7 +2275,7 @@ GameStore.Categories = {
           icons = { "Bath_Tub.png" },
           name = "Bath Tub",
           price = 250,
-          itemtype = 29312,
+          itemtype = 29336,
```
Comment (majestyotbr) [inline data/modules/scripts/gamestore/gamestore.lua]: Revert to 29312.

**Comment by andersonfaaria** [comment]
watch out for the warning in items.cpp:

src/items.cpp#L940
array index 2 is past the end of the array (which contains 2 elements) [-Warray-bounds]

**Comment by MajestyOTBR** [comment]
Can you see #1625? Thanks.

**Comment by Bipbop69** [comment]
when the player changes the npc's clothes it doesn't change the npc's clothes but the player's

edit-
ok

**Comment by FakeShinoda** [comment]
> 
> 
> when the player changes the npc's clothes it doesn't change the npc's clothes but the player's

outfit

Here it's working

**Comment by joao-blip** [comment]
> > when the player changes the npc's clothes it doesn't change the npc's clothes but the player's
> 
> outfit
> 
> Here it's working

can test with normal clothes

**Comment by attack787** [comment]
- When you try to change sex of Hireling, and you have his spawned in house, you get a mensage "Your hireling must be inside his/her lamp." You waste your coin and the sex isn't changed. Have a little drouble when you make confirmation of purchase, the mensage "purchase completed successfully" is shown two times interspersed with the option that choose what heriling will be changed sex.

Can you give a feedback about this bugs, maybe I did something wrong here

**Comment by FakeShinoda** [comment]
> 
> 
>     * When you try to change sex of Hireling, and you have his spawned in house, you get a mensage "Your hireling must be inside his/her lamp." You waste your coin and the sex isn't changed. Have a little drouble when you make confirmation of purchase, the mensage "purchase completed successfully" is shown two times interspersed with the option that choose what heriling will be changed sex.
> 
> 
> Can you give a feedback about this bugs, maybe I did something wrong here

Thanks for the report, i did not change the sex change method, i'll take a look into that.

**Review by majestyotbr — APPROVED** [review]

**Pull request merged**